### PR TITLE
Migrate to TypeScript

### LINKED ISSUE #71: Migrate the library to TypeScript
Remove Flow, and migrate to TS.

### PR DESCRIPTION
**Pull request by caseyWebb**
Closes #71 

- Replaces Flowtype with Typescript
- Nixes Babel (relies on TypeScript "target" setting in tsconfig.json)
- Adds necessary jest & eslint config

### Notes

- TS is stricter than Flowtype, specifically when it comes to null vs undefined. Currently I refactored to have no tangible differences, but if strict nulls are preferred I can make those changes.

- Some errors that had to be added b/c of the additional strictness of TS should have some meaningful messages added or otherwise handled

- There are some `any` types left in the hydra parser, but frankly I couldn't make sense of enough to come up with any that I trusted to be correct, and the ones I found published didn't seem to be what was really desired.

- I'd like to do a second pass refactor in a separate PR to clean up some of the promises. The only function refactored in this PR that wasn't strictly to satisfy Typescript was `fetchEntrypointAndDocs` in parseHydraDocumentation.ts because it made it way easier to reason about, and I made the change to Promise.all so the calls happen concurrently. Didn't want this PR to spiral out of control beyond what was necessary.

### CODE REVIEW

File: src/hydra/fetchJsonLd.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const jsonLdMimeType = "application/ld+json";
+
+/**
+ * Sends a JSON-LD request to the API.
+ *
+ * @param {string} url
+ * @param {object} options
+ * @return {Promise.<object>} An object with a response key (the original HTTP response) and an optional body key (the parsed JSON-LD body)
```
Comment (toofff) [inline src/hydra/fetchJsonLd.ts]: Fix `Promise.<object>` in `Promise<object>`, please

File: src/hydra/parseHydraDocumentation.ts [hunk 2]
```diff
@@ -188,7 +188,14 @@ function findRelatedClass(docs, property) {
  * @param {object} options
  * @return {Promise.<Api>}
```
Comment (toofff) [inline src/hydra/parseHydraDocumentation.ts]: Fix `Promise.<Api>` in `Promise<Api>`, please

File: src/hydra/fetchJsonLd.ts (hunk 1 above)
Comment (caseyWebb) [reply src/hydra/fetchJsonLd.ts]: I believe there is an argument to be made to remove the types entirely. I can't speak for other editors, but VS code will work with tsdoc annotations ootb.

See https://github.com/Microsoft/tsdoc.

Comment (dunglas) [reply src/hydra/fetchJsonLd.ts]: Let's remove useless docblock entirely then!

**Comment by soyuka** [comment]
nice work thanks @caseyWebb !

**Pull request merged**